lib/src/client/http2_connection.dart

```diff
@@ -344,24 +340,24 @@ class _SocketTransportConnector implements ClientTransportConnector {
           onBadCertificate: _validateBadCertificate);
     }
 
-    return ClientTransportConnection.viaSocket(_socket);
+    return ClientTransportConnection.viaSocket(_socket!);
   }
 
   @override
   String get authority =>
```

Comment:
I think this does not do anything useful when `_host` is not a String. So consider adding a `_host as String` at uses below.

Reply:
Some of the client tests started failing when I have replaced `toString` with `as`, so I have changed this to use `InternetAddress.host` when host is not a string